EL-3119 - Sankey Chart

ashley-hunter opened the pull request:
#### Checklist
<!-- Use an 'x' to check those which apply. -->
* [x] The contents of this PR are mine to submit. No third party code or other material is being committed.
* [x] New third party dependencies (if any) are linked via `package.json`. Third party dependencies are licenced under one of the following: Apache, MIT, BSD, Mozilla Public License, Eclipse Public License, or Oracle Binary Code License.
* [x] The code in this PR does not contain export-restricted encryption algorithms.

#### Ticket / Issue
<!-- Either a Jira URL or a description of the issue that this PR addresses. -->
https://portal.digitalsafe.net/browse/EL-3119

#### Description of Proposed Changes
<!-- Describe what was changed and how it addresses the original issue. -->
Adding Sankey Chart component, documentation and karma tests

#### Documentation CI URL
<!-- Initiate a build at https://jenkins.swinfra.net/job/SEPG/view/Templates/job/New%20SEPG%20Build/build -->
<!-- Append the branch name to the following URL: -->
https://pages.github.houston.softwaregrp.net/sepg-docs-qa/UXAspects_CI_UXAspects_EL-3119-Sankey-Chart

donna-marijne commented on package-lock.json:
```diff
@@ -11677,12 +11677,14 @@
               "bundled": true,
-              "dev": true
+              "dev": true,
+              "optional": true
```
Were these changes to package-lock.json intended?

ashley-hunter replied:
Reverted this, I'm working from home today so I needed to run an `npm install` which made these changes and they didn't need to be checked in

donna-marijne commented on src/components/sankey-chart/sankey-chart.component.html:
```diff
@@ -0,0 +1,79 @@
+        <stop class="ux-sankey-chart-falloff-gradient-start" offset="25%"></stop>
+        <stop class="ux-sankey-chart-falloff-gradient-end" offset="100%"></stop>
+      </linearGradient>
+    </defs>
```
Indentation misalignment here.

donna-marijne commented on src/components/sankey-chart/sankey-focus-manager.ts:
```diff
@@ -0,0 +1,124 @@
+import { DOWN_ARROW, LEFT_ARROW, RIGHT_ARROW, UP_ARROW } from '@angular/cdk/keycodes';
```
Perhaps in future (not this sprint) we can make a general purpose 2D focus manager service, since the algorithm for navigating within the sankey chart is rather similar to that in the partition map and dashboard.

ashley-hunter commented:
Added functionality to supply color for nodes & links via an input. Also fixed the indentation issue. Agreed, it would be good to standardize a way for navigating both horizontally and vertically with the keyboard.
